feat(loadout): adding jobs winter coats

i1yadobr opened the pull request:
Добавили уже существующие шубы отделов из билда в лодаут, чтобы игроки могли пользоваться ими на здоровье, и они не валялись как мёртвый груз.
Также добавили новое значение для gear_tweak'а, позволяющее создавать списки предметов, ориентирующихся на конкретную профессию.

<details>
<summary>Картинки</summary>

![4673635646747463](https://github.com/ChaoticOnyx/OnyxBay/assets/155672646/176c0b21-e35f-4ef7-bbb2-dba961041fb0)

![676785](https://github.com/ChaoticOnyx/OnyxBay/assets/155672646/5989772a-c19e-48a7-aba7-553fca37dd94)

</details>

<details>
<summary>Чейнджлог</summary>

```yml
🆑 ilyadobr & rufuszero
rscadd: Шубы отделов доступны в лодауте
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

i1yadobr committed:
adding new wintercoats in loadout

i1yadobr committed:
CMO issue

Filatelele commented:
Картинка действительно страшная. Есть мысль следующая. 

Добавить опцию Departmental winter coat которая будет автоматически назначать нужную шмотку.

i1yadobr commented:
Попробую потыкаться на днях с этим, если ничего не выйдет, буду просить подкрепление и помощи 😞

Commit:
feat(loadout): departmental gear

Add new "departmental" gear tweak
Apply departmental gear tweak to winter coats
Change behavior of character preview, low and medium job preferences are
used if high is not set
Add support for mutiple gear tweak options via "subtype" param

BaraBarax commented on code/modules/client/preference_setup/loadout/gear_tweaks.dm:
```diff
@@ -132,6 +132,74 @@
+	LAZYINITLIST(metadata)
+	var/datum/job/job = get_valid_current_job()
+	if(!istype(job))
+		return
```
```suggestion
		return
		
```

BaraBarax commented on code/modules/client/preference_setup/loadout/gear_tweaks.dm:
```diff
@@ -132,6 +132,74 @@
+
+/datum/gear_tweak/departmental/proc/get_valid_current_job()
+	if(current_job && (current_job.type in jobs_to_paths))
+		return current_job
```
```suggestion
		return current_job
		
```

BaraBarax commented on code/modules/client/preference_setup/loadout/gear_tweaks.dm:
```diff
@@ -132,6 +132,74 @@
+	if(current_job && (current_job.type in jobs_to_paths))
+		return current_job
+	if(!length(selected_jobs))
+		return
```
```suggestion
		return
		
```

BaraBarax commented on code/modules/client/preference_setup/loadout/loadout.dm:
```diff
@@ -667,6 +678,22 @@ var/list/hash_to_gear = list()
+/datum/gear/proc/is_departmental()
+	for(var/datum/gear_tweak/gt in gear_tweaks)
+		if(istype(gt, /datum/gear_tweak/departmental))
+			return TRUE
```
```suggestion
			return TRUE
			
```

rufuszero commented:
@BaraBarax без проблем, поправим, только можно уточнить, зачем эти ньюлайны нужны? Есть какой-то док по кодстайлу где с подобными предпочтениями мейнтейнеров и их причинами можно ознакомиться? Или это просто "так заведено"?

BaraBarax commented on code/modules/client/preference_setup/loadout/gear_tweaks.dm:
```diff
@@ -132,6 +132,74 @@
+	for(var/datum/job/job in selected_jobs)
+		var/list/choices = jobs_to_paths[job.type]
+		if(!(job.type in jobs_to_paths) || length(choices) < 2)
+			continue
```
```suggestion
			continue
			
```

BaraBarax commented on code/modules/client/preference_setup/loadout/loadout.dm:
```diff
@@ -366,6 +372,11 @@ var/list/hash_to_gear = list()
+					for(var/name in tweak_contents)
+						. += " <a href='?src=\ref[src];tweak=\ref[tweak];subtype=[tweak_contents[name]]'>[name]</a>"
+						. += "<br>"
+					continue
```
```suggestion
					continue
					
```

BaraBarax commented on code/modules/client/preference_setup/loadout/loadout.dm:
```diff
@@ -667,6 +678,22 @@ var/list/hash_to_gear = list()
+		CRASH("Expected list, got [selected_jobs]")
+	for(var/datum/gear_tweak/departmental/gt in gear_tweaks)
+		if(!istype(gt, /datum/gear_tweak/departmental))
+			continue
```
```suggestion
			continue
			
```

BaraBarax commented:
@rufuszero Гипотетически всё должно быть в [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md), но написано ли там про ньюлайны - хз

i1yadobr committed:
Lost spacebar

rufuszero commented:
@BaraBarax
Так я и смотрю в Guide to Contributing, я с ним полностью ознакомился перед заливкой своего кода.
![image](https://github.com/ChaoticOnyx/OnyxBay/assets/29821000/fa181740-6f64-42f7-bb22-9ed6ecb50579)
Просто не совсем понятно откуда такие моменты узнавать заранее чтоб не тратить время мейнтейнеров и контрибьюторов на ревьюшки буквально с ньюлайнами.

Filatelele commented:
Спасибо что указал на это. Добавлю, недоразумение вышло.

i1yadobr commented:
В связи с тем, что ПР не может пройти в билд, поскольку спрайты не соответствуют актуальному состоянию спрайтов билдов, ПР снова уходит в драфт, пока я не найду спрайтеров, которые допилят спрайты до актуального состояния, или я самолично этим не займусь.

i1yadobr committed:
Merge branch 'ChaoticOnyx:dev' into wintercoats

TobyThorne commented:
Полярку вернули, шубы больше не мёртвый груз. Редунтант ПР.
